dbgd.c

```diff
@@ -211,21 +211,55 @@ static int dbgd_mem_read(Dbg__Request *req, Dbg__Response *res)
     res->address = req->address;
     res->has_address = 1;
 
-    res->size = 1; /* FIXME: add word, dword, qword support */
-    res->has_size = 1;
+    res->data.len  = req->size;
+    res->data.data = malloc(res->data.len);
+    res->has_data = 1;
 
-    res->value = *((uint8_t*)(req->address));
-    res->has_value = 1;
+    unsigned int i = 0;
+    unsigned int s = req->size;
+
+    while(s >= 4) {
```

Comment:
Sorry for not getting back on this, I'm a bit busy with other things and forgot about this.

---

I have thought about this (and looked at interfaces) and your proposal is an unrealistic use case. If such behaviour is necessary, we can just extend our python set of functions with a `struct` like class which allows arbitrary access and groups multiple accesses in a single network packet which are executed in one "atomic" operation.

However, I do consider adding mem_read8(), mem_read16(), mem_read32() and remove any assumptions / restrictions of mem_read(), so for fast data transfers mem_read() can use a more optimized memcpy() or compression.